spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteViewCommands.scala

```diff
@@ -60,17 +65,27 @@ case class RewriteViewCommands(spark: SparkSession) extends Rule[LogicalPlan] wi
         properties = properties,
         allowExisting = allowExisting,
         replace = replace)
+
+    case a@AlterViewAs(ResolvedV2View(_, ident, _), _, query) =>
+      val q = CTESubstitution.apply(query)
+      verifyTemporaryObjectsDontExist(ident, q)
+      a.copy(query = q)
+
+    case u@UnresolvedView(IsIcebergView(resolved), _, _, _) =>
+      loadView(resolved.catalog, resolved.identifier)
+        .map(v => ResolvedV2View(resolved.catalog.asViewCatalog, resolved.identifier, v))
+        .getOrElse(u)
+
+    case ShowViews(UnresolvedNamespace(ns), pattern, output)
+      if isViewCatalog(spark.sessionState.catalogManager.currentCatalog) =>
+      ShowIcebergViews(UnresolvedNamespace(ns), pattern, output)
```

Comment:
Shouldn't this resolve the namespace first? This could be called for cases like `SHOW VIEWS IN prod.db`, where the `UnresolvedNamespace` has both `prod` (a catalog) and `db` (a namespace). This needs to run `CatalogAndNamespace.unapply` to get the correct catalog and namespace.

Reply:
I've updated this to use `CatalogAndNamespace` for this particular case. For the case where no namespace is given, we need to match `ShowViews(UnresolvedNamespace(Seq()), pattern, output)`